src/legacy/core_plugins/timelion/public/plugin.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  CoreSetup,
+  CoreStart,
+  InternalCoreStart,
+  Plugin,
+  PluginInitializerContext,
+} from 'kibana/public';
+import { VisualizationsSetup } from 'src/legacy/core_plugins/visualizations/public';
+import { Plugin as DataPublicPlugin } from 'src/plugins/data/public';
+import { IFeatureCatalogueRegistryProvider } from 'ui/registry/feature_catalogue';
+// @ts-ignore
+import { timelionVis } from './timelion_vis_fn';
```

Comment:
Could you please also migrate files to TS the following files: 
timelion_vis_fn, vis, register_feature, timechart

Reply:
Not sure. It impacts other files more than this one.

Reply:
Except mentioned ones some files also have been moved. Thanks